### PR TITLE
Adds v0.2 Release Announcement Doc

### PR DESCRIPTION
**Pull request by danehans**
Adds a doc for announcing the v0.2 release.

Signed-off-by: danehans <daneyonhansen@gmail.com>

### CODE REVIEW

File: docs/releases/v0.2.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+### Kubernetes Support
+
+Run Envoy Gateway in a Kubernetes cluster. Checkout the [quickstart guide][] to get started with Envoy Gateway in a few
+simple steps.
+
+### Gateway API Support
+
+Envoy Gateway supports Gateway API resources for running and configuring a managed fleet of Envoy proxies. Envoy Gateway
+passes Gateway API core [conformance tests][] and supports GatewayClass, Gateway, HTTPRoute, TLSRoute, and
+ReferenceGrant resources. See the [documentation][docs] for additional details on how to use Envoy Gateway for your edge
```
Comment (arkodg) [inline docs/releases/v0.2.md:36]: we should rm `ReferenceGrant` from here since the support for it is incomplete

**Review by arkodg — APPROVED** [review]

**Review by LukeShu — COMMENTED** [review]
2 issues: Arko's comment, and also the spelling mistake found by the linter.

Otherwise, lgtm

**Comment by danehans** [comment]
>2 issues: Arko's comment, and also the spelling mistake found by the linter.

Both were fixed by commit [6275af4](https://github.com/envoyproxy/gateway/pull/635/commits/6275af4dd7a16c5e11dd04e95abdc2eefa51486f).

**Comment by danehans** [comment]
Rerunning CI due to possible flake:
```
--- FAIL: TestProvider (11.46s)
    --- FAIL: TestProvider/gatewayclass_accepted_status (0.10s)
        kubernetes_test.go:168: 
            	Error Trace:	/home/runner/work/gateway/gateway/internal/provider/kubernetes/kubernetes_test.go:168
            	            				/home/runner/work/gateway/gateway/internal/provider/kubernetes/kubernetes_test.go:78
            	Error:      	Not equal: 
            	            	expected: &v1beta1.GatewayClass{TypeMeta:v1.TypeMeta{Kind:"GatewayClass", APIVersion:"gateway.networking.k8s.io/v1beta1"}, ObjectMeta:v1.ObjectMeta{Name:"test-gc-accepted-status", GenerateName:"", Namespace:"", SelfLink:"", UID:"c5155bf2-d8f0-4717-a55a-2f7c4f7e4fe7", ResourceVersion:"288", Generation:1, CreationTimestamp:time.Date(2022, time.October, 20, 19, 22, 2, 0, time.Local), DeletionTimestamp:<nil>, DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string(nil), Annotations:map[string]string(nil), OwnerReferences:[]v1.OwnerReference(nil), Finalizers:[]string(nil), ZZZ_DeprecatedClusterName:"", ManagedFields:[]v1.ManagedFieldsEntry{v1.ManagedFieldsEntry{Manager:"kubernetes.test", Operation:"Update", APIVersion:"gateway.networking.k8s.io/v1beta1", Time:time.Date(2022, time.October, 20, 19, 22, 2, 0, time.Local), FieldsType:"FieldsV1", FieldsV1:(*v1.FieldsV1)(0xc000bb6fa8), Subresource:""}, v1.ManagedFieldsEntry{Manager:"kubernetes.test", Operation:"Update", APIVersion:"gateway.networking.k8s.io/v1beta1", Time:time.Date(2022, time.October, 20, 19, 22, 2, 0, time.Local), FieldsType:"FieldsV1", FieldsV1:(*v1.FieldsV1)(0xc000bb6fd8), Subresource:"status"}}}, Spec:v1beta1.GatewayClassSpec{ControllerName:"gateway.envoyproxy.io/gatewayclass-controller", ParametersRef:(*v1beta1.ParametersReference)(nil), Description:(*string)(nil)}, Status:v1beta1.GatewayClassStatus{Conditions:[]v1.Condition{v1.Condition{Type:"Accepted", Status:"True", ObservedGeneration:1, LastTransitionTime:time.Date(2022, time.October, 20, 19, 22, 2, 0, time.Local), Reason:"Accepted", Message:"Valid GatewayClass"}}}}
            	            	actual  : &v1beta1.GatewayClass{TypeMeta:v1.TypeMeta{Kind:"GatewayClass", APIVersion:"gateway.networking.k8s.io/v1beta1"}, ObjectMeta:v1.ObjectMeta{Name:"test-gc-accepted-status", GenerateName:"", Namespace:"", SelfLink:"", UID:"c5155bf2-d8f0-4717-a55a-2f7c4f7e4fe7", ResourceVersion:"287", Generation:1, CreationTimestamp:time.Date(2022, time.October, 20, 19, 22, 2, 0, time.Local), DeletionTimestamp:<nil>, DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string(nil), Annotations:map[string]string(nil), OwnerReferences:[]v1.OwnerReference(nil), Finalizers:[]string(nil), ZZZ_DeprecatedClusterName:"", ManagedFields:[]v1.ManagedFieldsEntry{v1.ManagedFieldsEntry{Manager:"kubernetes.test", Operation:"Update", APIVersion:"gateway.networking.k8s.io/v1beta1", Time:time.Date(2022, time.October, 20, 19, 22, 2, 0, time.Local), FieldsType:"FieldsV1", FieldsV1:(*v1.FieldsV1)(0xc000bb7008), Subresource:""}}}, Spec:v1beta1.GatewayClassSpec{ControllerName:"gateway.envoyproxy.io/gatewayclass-controller", ParametersRef:(*v1beta1.ParametersReference)(nil), Description:(*string)(nil)}, Status:v1beta1.GatewayClassStatus{Conditions:[]v1.Condition{v1.Condition{Type:"Accepted", Status:"Unknown", ObservedGeneration:0, LastTransitionTime:time.Date(1970, time.January, 1, 0, 0, 0, 0, time.Local), Reason:"Waiting", Message:"Waiting for controller"}}}}
```

**Pull request merged**